shipyard-controller/handler/projectmanager.go

```diff
@@ -472,6 +503,31 @@ func toModelProject(project models.ExpandedProject) apimodels.Project {
 	}
 }
 
+func (pm *ProjectManager) validateShipyardStagesUnchaged(oldProject *models.ExpandedProject, newProject *models.ExpandedProject) error {
+	if len(newProject.Stages) != len(oldProject.Stages) {
+		return fmt.Errorf("unallowed addition/removal of project stages")
+	}
+
+	for i, oldStage := range oldProject.Stages {
+		if oldStage.StageName != newProject.Stages[i].StageName {
+			if !stageInArrayOfStages(oldStage.StageName, newProject.Stages) {
```

Comment:
This check covers also the one at line 512.

Reply:
Thank you for the nice explanation, I would add a comment to explain this briefly otherwise while scheming through the code it is not immediate why this check is done :)

Reply:
I will create a PR with a comment describing the reason for the double check. Thanks for the review :)